torchvision/models/vision_transformer.py

```diff
@@ -135,6 +145,7 @@ def __init__(
         num_classes: int = 1000,
         representation_size: Optional[int] = None,
         norm_layer: Callable[..., torch.nn.Module] = partial(nn.LayerNorm, eps=1e-6),
+        conv_stem_layers: Union[List[ConvStemLayer], List[Dict], None] = None,
```

Comment:
I think we should choose to either accept the named tuple or the dictionary, not both. This will lead to a simpler API. I think the idea of using a config object is a good approach comparing to a Dictionary because it clearly defines the supported supported options:
```suggestion
        conv_stem_layers: Optional[List[ConvStemLayer]] = None,
```